### PR TITLE
fix: add javadoc for `ApiFutures`

### LINKED ISSUE #1239: Missing Javadoc over methods in ApiFutures
Hello,

In one of our projects we're using some methods from the [`ApiFutures`](https://github.com/googleapis/api-common-java/blob/main/src/main/java/com/google/api/core/ApiFutures.java) class, which basically provide implementations of the methods from Guava's [`Futures`](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/Futures.java) class, but accept `ApiFuture`s, instead of `ListenableFuture`s.  Most (all?) of the implementations in `ApiFutures` are just simple delegators to the equivalent methods in the `Futures` class, after they wrapped the `ApiFuture` argument in a `ListenableFuture` object.

However, when we tried to understand the behavior of the methods in `ApiFutures`, without looking at their implementations, we noticed that they lack Javadoc, not even simple `@see` references to the equivlaent method in the `Futures` class. People who are not familiar with Java's `Future`s, and read code that uses methods from `ApiFutures` for the first time, struggle to understand their semantics without looking at the implementations.

Can you please add the missing Javadoc over the methods in this class?

Thanks!

### PR DESCRIPTION
**Pull request by JoeWang1127**
Fixes #1239 ☕️

### CODE REVIEW

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java [hunk 1]
```diff
@@ -159,57 +232,89 @@ public static <V, X> ApiFuture<X> transform(
...
     return new ListenableFutureToApiFuture<>(
         Futures.allAsList(
             Iterables.transform(
                 futures,
-                new Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>() {
-                  public ListenableFuture<? extends V> apply(ApiFuture<? extends V> apiFuture) {
-                    return listenableFutureForApiFuture(apiFuture);
-                  }
-                })));
+                (Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>)
```
Comment (burkedavison) [inline api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: Please remove the cast. It shouldn't be needed.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java [hunk 2]
```diff
@@ -159,57 +232,89 @@ public static <V, X> ApiFuture<X> transform(
...
-                  }
-                })));
-  }
-
-  /*
-   * @deprecated Use {@linkplain #transformAsync(ApiFuture, ApiFunction, Executor) the
-   * overload that requires an executor}. For identical behavior, pass {@link
-   * com.google.common.util.concurrent.MoreExecutors#directExecutor}, but consider whether
-   * another executor would be safer.
+                (Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>)
```
Comment (burkedavison) [inline api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: Please remove the cast. It shouldn't be needed.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java [hunk 3]
```diff
@@ -45,18 +45,30 @@
...
-   * another executor would be safer.
+  /**
+   * @deprecated Use {@linkplain #addCallback(ApiFuture, ApiFutureCallback, Executor) the overload
+   *     that requires an executor}. For identical behavior, pass {@link
+   *     com.google.common.util.concurrent.MoreExecutors#directExecutor}, but consider whether
+   *     another executor would be safer.
    */
   @Deprecated
   public static <V> void addCallback(
       final ApiFuture<V> future, final ApiFutureCallback<? super V> callback) {
```
Comment (burkedavison) [inline api-common-java/src/main/java/com/google/api/core/ApiFutures.java:56]: One major intent of this ticket is to improve the reference API documentation found at https://cloud.google.com/java/docs/reference/api-common/latest/com.google.api.core.ApiFutures, so please make sure that each of these methods have entries for `@param`, `@return`, and `@throws` as appropriate.

**Review by burkedavison — CHANGES_REQUESTED** [review]

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java (hunk 2 above)
Comment (JoeWang1127) [reply api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: This is outdated.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java (hunk 1 above)
Comment (JoeWang1127) [reply api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: This is outdated.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java (hunk 3 above)
Comment (JoeWang1127) [reply api-common-java/src/main/java/com/google/api/core/ApiFutures.java:61]: done.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java [hunk 4]
```diff
@@ -45,18 +45,35 @@
 public final class ApiFutures {
   private ApiFutures() {}
 
-  /*
-   * @deprecated Use {@linkplain #addCallback(ApiFuture, ApiFutureCallback, Executor) the
-   * overload that requires an executor}. For identical behavior, pass {@link
-   * com.google.common.util.concurrent.MoreExecutors#directExecutor}, but consider whether
-   * another executor would be safer.
+  /**
+   * Registers separate success and failure callbacks to be run when the {@link ApiFuture}'s
```
Comment (blakeli0) [inline api-common-java/src/main/java/com/google/api/core/ApiFutures.java:49]:
```suggestion
   * Registers a callback to be run when the {@link ApiFuture}'s
```
It's just one callback being passed in, the original doc could confuse people that there are multiple callbacks passed in.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java [hunk 5]
```diff
@@ -45,18 +45,35 @@
...
+   *     another executor would be safer.
    */
   @Deprecated
   public static <V> void addCallback(
       final ApiFuture<V> future, final ApiFutureCallback<? super V> callback) {
     addCallback(future, callback, directExecutor());
   }
 
+  /**
+   * Registers separate success and failure callbacks to be run when the {@link ApiFuture}'s
```
Comment (blakeli0) [inline api-common-java/src/main/java/com/google/api/core/ApiFutures.java:66]:
```suggestion
   * Registers a callback to be run when the {@link ApiFuture}'s
```
Same as the above.

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java (hunk 4 above)
Comment (JoeWang1127) [reply api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: done

File: api-common-java/src/main/java/com/google/api/core/ApiFutures.java (hunk 5 above)
Comment (JoeWang1127) [reply api-common-java/src/main/java/com/google/api/core/ApiFutures.java]: done

**Review by blakeli0 — APPROVED** [review]

**Comment by burkedavison** [comment]
LGTM

**Review by burkedavison — APPROVED** [review]

**Pull request merged**